src/test/scala/org/opennetworkinsight/LDAWrapperTest.scala

```diff
@@ -0,0 +1,153 @@
+package org.opennetworkinsight
+
+import org.opennetworkinsight.testutils.TestingSparkContextFlatSpec
+import org.scalatest.Matchers
+
+
+class LDAWrapperTest extends TestingSparkContextFlatSpec with Matchers{
+
+
+  "normalizeWord" should "calculate exponential of each value in the input string, then sum up all the exponential and " +
+    "then divided  each exponential by the total sum" in {
+
+    val wordProbability = "1 2 3 4 5"
+    val result = LDAWrapper.normalizeWord(wordProbability)
+
+    result.size shouldBe 5
+    result(0) shouldBe 0.011656230956039607
+    result(1) shouldBe 0.03168492079612427
+    result(2) shouldBe 0.0861285444362687
+    result(3) shouldBe 0.23412165725273662
+    result(4) shouldBe 0.6364086465588308
+  }
+
+  "getTopicDocument" should "return string of document and 20 values when the sum of each value in the line is bigger" +
+    "than 0. Each result value should be divided by the sum of all values" in {
+
+    val document = "192.168.1.1"
+    val line = "0.0124531442 0.0124531442 0.0124531442 0.0124531442 0.0124531442 0.0124531442 0.0124531442 0.0124531442 " +
+      "0.0124531442 0.0124531442 0.0124531442 23983.5532262138 0.0124531442 0.0124531442 0.0124531442 0.0124531442 " +
+      "0.0124531442 0.0124531442 22999.4716800747 0.0124531442"
+
+    var result = LDAWrapper.getTopicDocument(document, line)
+
+    result shouldBe "192.168.1.1,2.6505498126219955E-7 2.6505498126219955E-7 2.6505498126219955E-7 2.6505498126219955E-7" +
+      " 2.6505498126219955E-7 2.6505498126219955E-7 2.6505498126219955E-7 2.6505498126219955E-7 2.6505498126219955E-7" +
+      " 2.6505498126219955E-7 2.6505498126219955E-7 0.5104702996191969 2.6505498126219955E-7 2.6505498126219955E-7" +
+      " 2.6505498126219955E-7 2.6505498126219955E-7 2.6505498126219955E-7 2.6505498126219955E-7 0.48952492939114034" +
+      " 2.6505498126219955E-7"
+  }
+
+  it should "return string of document and 0.0 20 times when the sum of each value in the line is 0" in {
+    val document = "192.168.1.1"
+    val line = "0.0 0.0 1.0 -1.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0"
+
+    val result = LDAWrapper.getTopicDocument(document, line)
+
+    result shouldBe "192.168.1.1,0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0"
+  }
+
+  "createModel" should "return model in Array[String] format. Each string should contain the document count and the" +
+    "total count for each word" in {
+
+    val documentWordData = sparkContext.parallelize(Array(Array("192.168.1.1", "333333_7.0_0.0_1.0", "8"),
+      Array("10.10.98.123", "1111111_6.0_3.0_5.0", "4"),
```

Comment:
this isn't the originating location in the codebase... but low priority item (but a good one for readability) is that the binning numbers should be INTs not DOUBLEs for the word creation... the extra  ".0"s throughout the word strings hinder readability